src/util/hass-translation.ts

```diff
@@ -39,18 +42,27 @@ function findAvailableLanguage(language: string) {
 }
 
 /**
- * Get user selected language from backend
+ * Get user selected locale data from backend
  */
-export async function getUserLanguage(hass: HomeAssistant) {
+export async function getUserLocale(
+  hass: HomeAssistant
+): Promise<FrontendTranslationData | null> {
   const result = await fetchTranslationPreferences(hass);
-  const language = result ? result.language : null;
-  if (language) {
+  const language = result?.language || hass.locale.language;
+  const number_format = result?.number_format || hass.locale.number_format;
```

Comment:
```suggestion
  const language = result?.language;
  const number_format = result?.number_format;
```

Reply:
@bramkragten these changes will result in type errors, because `getUserLocale()` returns `Promise<FrontendTranslationData>` and `language` and `number_format` could now be `undefined` which is not a valid value for either of those properties. I can update the `FrontendTranslationData` interface to allow for `null` or `undefined` values, is that how you'd prefer to handle the errors? Generally I avoid allowing those on type interfaces if possible, which is why I wrote in the fallback values.

The `hassConnected()` function checks for falsy values before called `_selectLanguage()` and `_selectNumberFormat()` so updating the interface should work to resolve the type errors.

Reply:
In that case, I would use a different interface here, or `Partial<FrontendTranslationData>`